### ISSUE TITLE
Mixing watcher with useSignificantChanges and getCurrentPosition

### ISSUE THREAD

**Issue by cristianoccazinsp** [react-native-community/react-native-geolocation#85]
## Environment

```
System:
    OS: macOS Mojave 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-6267U CPU @ 2.90GHz
    Memory: 20.18 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.15.3 - /usr/local/bin/node
    npm: 6.13.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 28
      Build Tools: 28.0.3
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5522156
    Xcode: 11.2.1/11B53 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.12.0 => 16.12.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    create-react-native-app: 2.0.2
    create-react-native-modu
```

## Platforms

iOS

## Versions

- iOS: 13.2.3
- react-native-geolocation: 2.0.2
- react-native: 0.61.5
- react: 16.12.0

## Description

Looks like adding a watcher with `useSignificantChanges` true and then calling `getCurrentPosition` (without significant changes) will always make the request fail because of the watcher. The internal iOS code only manages 1 single instance of the location manager object and if a watcher is set with significant changes, all location requests will timeout because it won't receive new data.

How would we fix this? Any work around?

## Reproducible Demo
-

**Comment by cristianoccazinsp** [comment]
Just a little update about what the issue is:

Let's say we want to call getCurrentPosition to be precise (and without significant location), but we also want to monitor location with a watch using significant location changes.

We create a watch with the significant location changes flag, and later on we call `getCurrentPosition` without it. The final result is that our watch now lost the significant location changes flag and will request full GPS data.

**Comment by JorenVos** [comment]
I'm currently dealing with exactly the same problem. When watching for location updates in the background, it's impossible to use `getCurrentPosition` in another part of the application.

**Comment by cristianoccazinsp** [comment]
I've tried to tackle the issue in this PR (https://github.com/react-native-community/react-native-geolocation/pull/88) by using 2 managers instead of 2; one for the watcher and one for getCurrentPosition.  For my use-case it is working fine, but I don't know for sure if the change will help everyone, do you mind trying it?

**Comment by OlivierVoyer** [comment]
It looks like I have observed some similar problems due to the same root cause of having only one CLLocationManager internally : https://github.com/Agontuk/react-native-geolocation-service/issues/172

**Comment by acoulon99** [comment]
Not ideal, but could stopping and restarting the watch serve as a workaround? 

```
  function getCurrentPosition() {
    Geolocation.stopObserving()
    Geolocation.watchPosition(updateLocation, handleError, options)
  }
```

**Comment by sahilpandya** [comment]
@here Any update or alternate to fix watchLocation and getCurrentPosition usage conflicts issue.

**Comment by JorenVos** [comment]
> @here Any update or alternate to fix watchLocation and getCurrentPosition usage conflicts issue.

I'm using https://github.com/mauron85/react-native-background-geolocation for "watchLocation" purposes as it handles background watching and https://github.com/Agontuk/react-native-geolocation-service for "getCurrentPosition" purposes. Both of them are doing the job quite well.

Background geolocation might be a bit harder to setup and require some on-the-field testing to get everything right for both iOS and Android, I ended up having good location tracking on iOS, while having too late update intervals on Android. Easy fix was available by setting the distanceFilter :) I ended up having this config in code (for more info, check @mauron85/react-native-background-geolocation): 

```
BackgroundGeolocation.configure({
            desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
            stationaryRadius: 50,
            distanceFilter: 10,
            notificationTitle: 'title',
            notificationText: 'message',
            notificationIconSmall: 'icon',
            notificationIconColor: ''#00FF00,
            interval: 1000,
            fastestInterval: 1000,
            locationProvider: BackgroundGeolocation.RAW_PROVIDER,
        });
```

**Comment by sahilpandya** [comment]
> > @here Any update or alternate to fix watchLocation and getCurrentPosition usage conflicts issue.
> 
> I'm using https://github.com/mauron85/react-native-background-geolocation for "watchLocation" purposes as it handles background watching and https://github.com/Agontuk/react-native-geolocation-service for "getCurrentPosition" purposes. Both of them are doing the job quite well.
> 
> Background geolocation might be a bit harder to setup and require some on-the-field testing to get everything right for both iOS and Android, I ended up having good location tracking on iOS, while having too late update intervals on Android. Easy fix was available by setting the distanceFilter :) For me, the configuration docs were not clear enough at some points. I ended up having this config in code (for more info, check @mauron85/react-native-background-geolocation), if you want to use it for high accuracy location tracking:
> 
> ```
> import BackgroundGeolocation from '@mauron85/react-native-background-geolocation';
> 
> BackgroundGeolocation.configure({
>             desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
>             stationaryRadius: 50,
>             distanceFilter: 10,
>             notificationTitle: 'title',
>             notificationText: 'message',
>             notificationIconSmall: 'icon',
>             notificationIconColor: ''#00FF00,
>             interval: 1000,
>             fastestInterval: 1000,
>             locationProvider: BackgroundGeolocation.RAW_PROVIDER,
>         });
> ```

Yeah, I fixed above problem writing own native Geo file for watcher.
Here is a sample:
`RCT_EXPORT_METHOD(startLocationUpdates)
{
  [self initialize];
  [_locationManager startMonitoringSignificantLocationChanges];
  [_locationManager startUpdatingLocation];
}
`

**Comment by lyseiha** [comment]
Hello @JorenVos, would you mind to sharing the code snippet of the location tracking when app killed/terminated?
